### PR TITLE
fix(condo): DOMA-4416 fixed multiple filter modal

### PR DESCRIPTION
**Pull request by Alllex202**
Problems: 
 - was reseted selected templates,
 - incorrect operation of filter reset
 
 Solution: 
  - 'MultipleFilterContextProvider' saving selected filters template without passing props (It doesn't cause unnecessary rendering)
  - Reverting "MultipleFiltersModal" to "useCallback" allows the display of "MultipleFiltersModal" to properly update when visibility changes

### CODE REVIEW

**Review by pahaz — APPROVED** [review]

File: apps/condo/domains/common/hooks/useMultipleFiltersModal.tsx [hunk 1]
```diff
@@ -52,6 +52,8 @@ import {
     updateQuery,
 } from '../utils/filters.utils'
 import { Ticket } from '@app/condo/schema'
+import { has, isNil, omitBy } from 'lodash'
```
Comment (akarjakin) [inline apps/condo/domains/common/hooks/useMultipleFiltersModal.tsx:55]: please use individual imports from lodash in .tsx

Comment (akarjakin) [inline apps/condo/domains/common/hooks/useMultipleFiltersModal.tsx:55]: we use to sort imports

File: apps/condo/domains/common/hooks/useMultipleFiltersModal.tsx [hunk 2]
```diff
@@ -298,6 +300,29 @@ function getModalComponents <T> (filters: IFilters, filterMetas: Array<FiltersMe
...
+
+export const useMultipleFilterContext = (): IFilterContext => useContext<IFilterContext>(FilterContext)
+
+export const MultipleFilterContextProvider: React.FC = ({ children }) => {
+    const [selectedFiltersTemplate, setSelectedFiltersTemplate] = useState()
+
+    return (
+        <FilterContext.Provider
+            children={children}
+            value={{
```
Comment (akarjakin) [inline apps/condo/domains/common/hooks/useMultipleFiltersModal.tsx:318]: better to memoize this object

File: apps/condo/pages/meter/index.tsx [hunk 3]
```diff
@@ -254,14 +257,16 @@ const MetersPage: IMeterIndexPage = () => {
...
-            sortBy={sortersToSortBy(sorters) as SortMeterReadingsBy[]}
-            filterMetas={filterMetas}
-            role={role}
-            loading={isLoading}
-        />
+        <MultipleFilterContextProvider>
+            <MetersPageContent
+                tableColumns={tableColumns}
+                searchMeterReadingsQuery={searchMeterReadingsQuery}
+                sortBy={sortersToSortBy(sorters) as SortMeterReadingsBy[]}
```
Comment (akarjakin) [inline apps/condo/pages/meter/index.tsx:264]: useMemo

**Pull request merged**